data-pipeline/src/pipeline/pre_processing/ancillary/local_authority.py

```diff
@@ -28,18 +30,30 @@ def build_local_authorities(
         outturn_filepath_or_buffer,
         year,
     )
-
-    logger.info("Processing Local Authority statistical neighbours data.")
-
-    la_statistical_neighbours_data = _prepare_la_statistical_neighbours(
+    ons_la_population_data = _prepare_ons_la_population_data(
+        ons_filepath_or_buffer,
+        year,
+    )
+    statistical_neighbours_data = _prepare_la_statistical_neighbours(
         statistical_neighbours_filepath, year
     )
 
+    logger.info("Processing Local Authority combined data.")
+
     local_authority_data = section_251_data.merge(
-        la_statistical_neighbours_data,
+        statistical_neighbours_data,
         left_on="old_la_code",
         right_index=True,
         how="left",
+    ).merge(
+        ons_la_population_data,
+        left_on="new_la_code",
+        right_index=True,
+        how="left",
+    )
```

Comment:
Actually, thinking about it, this could probably do with further processing:

```python
.reset_index()
.drop(columns=["new_la_code"])
.rename(columns={"old_la_code": "Code"})
.set_index("Code")
```

…to keep it in line with the data we know we're going to need.

Reply:
Ignore: we'll pick this up as part of the DB ingestion piece.